rust/query-stats/README.md

```diff
@@ -0,0 +1,75 @@
+# Query stats
+
+## Prerequisites
+This example requires an installation of:
+
+- [x] Install the [IC SDK](../developer-docs/setup/install/index.mdx) at version 0.16.1 or higher.
+- [x] Download the following project files from GitHub: https://github.com/dfinity/examples/
+
+Begin by opening a terminal window.
+
+This is currently only for use from within DFINITY, as the replica shipped with `dfx` as well as nodes on 
+mainnet do not enable the query stats feature at the moment.
+
+Once enabled on mainnet, the following instructions will work without the `--network` argument.
+
+From within DFINITY, it can be ran as following (e.g. with a deployment to the `small_with_query_stats` Farm testnet)
+
+Variable `$BN_HOSTNAME` points at the hostname of a boundary node. For example, from field `['bn_aaaa_records']['url']` of the output of a Farm deployment
+
+## Installing the canister
+
+```
+dfx canister create --network="$BN_HOSTNAME" query_stats
+```
+
+We will need the canister ID that has been assigned to our canister later.
+
+```
+dfx build --network=$BN_HOSTNAME query_stats
+dfx canister install --network=$BN_HOSTNAME query_stats
+```
+
+## Issuing load
+
+We now want to generate queries at a certain rate. One way to achieve this is by using `ic-workload-generator`, which is part of the [IC repo](https://github.com/dfinity/ic/tree/master/rs/workload_generator). 
+
+
+```
+ic-workload-generator $BN_HOSTNAME -r 10 -n 6000 --call-method "load" -m query --payload "4449444c0000" --no-status-check --canister-id=$CANISTER_ID
+```
+
+
+## Get query stats
+
+There is a delay until query stats show up in the canister status. Depending on the configuration, this 
+can reach from multiple minutes to an hour.
```

Comment:
How does this behave on local? Is it easily testable there?

Reply:
Not currently, but as discussed offline, we should probably enable it in the `dfx` replica. I will talk to people to come up with a suggestion on how to do that.